### PR TITLE
Add ability to create games sign in

### PR DESCRIPTION
**Pull request by ihildebrandt**
This introduces an additional `init` argument so that a games sign in experience can be created in Android. The native iOS code will return an error if attempting to create a games sign in. 

Additionally, the Android plugin publishes the `GoogleSignInAccount` so that it can be accessed by other native code.

### CODE REVIEW

File: packages/google_sign_in/android/src/main/java/io/flutter/plugins/googlesignin/GoogleSignInPlugin.java [hunk 1]
```diff
@@ -380,6 +403,7 @@ private void onSignInResult(GoogleSignInResult result) {
       if (result.isSuccess()) {
         GoogleSignInAccount account = result.getSignInAccount();
         currentAccount = account;
+        registrar.publish(account);
```
Comment (mravn-google) [inline packages/google_sign_in/android/src/main/java/io/flutter/plugins/googlesignin/GoogleSignInPlugin.java]: We were planning to deprecate the value publishing API. Can't potential clients use the GoogleSignIn API directly to get the account?

File: packages/google_sign_in/android/src/main/java/io/flutter/plugins/googlesignin/GoogleSignInPlugin.java [hunk 2]
```diff
@@ -206,14 +211,30 @@ private void checkAndSetPendingOperation(String method, Result result) {
...
       // We're not initialized until we receive `onConnected`.
       // If initialization fails, we'll receive `onConnectionFailed`
       checkAndSetPendingOperation(METHOD_INIT, result);
 
       try {
-        GoogleSignInOptions.Builder optionsBuilder =
-            new GoogleSignInOptions.Builder(GoogleSignInOptions.DEFAULT_SIGN_IN).requestEmail();
+        GoogleSignInOptions.Builder optionsBuilder;
+
+        if (builderOptions == null) builderOptions = DEFAULT_SIGN_IN;
```
Comment (mravn-google) [inline packages/google_sign_in/android/src/main/java/io/flutter/plugins/googlesignin/GoogleSignInPlugin.java]: This should not be necessary, as we ensure the parameter has a non-null value on the Dart side.

**Comment by ihildebrandt** [comment]
@mravn-google Thanks for the review. I believe I addressed all of your notes. Swapping to a direct API call in my downstream plugin worked. 

Let me know if there's anything else I can clean up.

File: packages/google_sign_in/ios/Classes/GoogleSignInPlugin.m [hunk 3]
```diff
@@ -67,17 +67,25 @@ - (instancetype)init {
...
-      NSMutableDictionary *plist = [[NSMutableDictionary alloc] initWithContentsOfFile:path];
-      [GIDSignIn sharedInstance].clientID = plist[kClientIdKey];
-      [GIDSignIn sharedInstance].scopes = call.arguments[@"scopes"];
-      [GIDSignIn sharedInstance].hostedDomain = call.arguments[@"hostedDomain"];
-      result(nil);
-    } else {
-      result([FlutterError errorWithCode:@"missing-config"
-                                 message:@"GoogleService-Info.plist file not found"
+    if ([((NSNumber *)call.arguments[@"signInOption"]) isEqualToNumber:[NSNumber
+                                                                           numberWithInt:1]]) {
```
Comment (mravn-google) [inline packages/google_sign_in/ios/Classes/GoogleSignInPlugin.m]: Maybe simpler with
```objectivec
NSNumber* signInOption = call.arguments[@"signInOption"];
if ([signInOption intValue] == 1) {
  ...
}
```

File: packages/google_sign_in/lib/google_sign_in.dart [hunk 4]
```diff
@@ -127,6 +129,9 @@ class GoogleSignIn {
   static const MethodChannel channel =
       const MethodChannel('plugins.flutter.io/google_sign_in');
 
+  /// The options to pass to GoogleSignInOptions.Builder()
```
Comment (mravn-google) [inline packages/google_sign_in/lib/google_sign_in.dart]: We should not mention the Android API's builder here, but instead document what `SignInOption` is, in its own right. And we should document that `gamesSignIn` isn't supported on iOS.

File: packages/google_sign_in/lib/google_sign_in.dart [hunk 5]
```diff
@@ -142,7 +147,21 @@ class GoogleSignIn {
...
+  /// Factory for creating default sign in
+  factory GoogleSignIn.defaultSignIn(
+      {List<String> scopes, String hostedDomain}) {
+    return new GoogleSignIn(
+        signInOption: SignInOption.defaultSignIn,
+        scopes: scopes,
+        hostedDomain: hostedDomain);
+  }
+
+  /// Factory for creating sign in suitable for games
```
Comment (mravn-google) [inline packages/google_sign_in/lib/google_sign_in.dart]: Please document that this does not work on iOS.

File: packages/google_sign_in/lib/google_sign_in.dart [hunk 6]
```diff
@@ -13,6 +13,8 @@ import 'src/common.dart';
 export 'src/common.dart';
 export 'widgets.dart';
 
+enum SignInOption { defaultSignIn, gamesSignIn }
```
Comment (mravn-google) [inline packages/google_sign_in/lib/google_sign_in.dart]: Maybe call these `standard` and `games`. The `SignIn` part is already given by the enum type name.

**Comment by mravn-google** [comment]
Thanks a lot! Can I ask you to address the formatting discrepancies [reported by Travis](https://travis-ci.org/flutter/plugins/jobs/373286752)?

File: packages/google_sign_in/ios/Classes/GoogleSignInPlugin.m [hunk 7]
```diff
@@ -65,26 +66,32 @@ - (instancetype)init {
...
+- (void)handleMethodCall:(FlutterMethodCall *)call
+                  result:(FlutterResult)result {
   if ([call.method isEqualToString:@"init"]) {
-    NSNumber *signInOption = call.arguments[@"signInOption"];
-    if ([signInOption intValue] == 1) {
-      result([FlutterError errorWithCode:@"unsupported-options"
-                                 message:@"Games sign in is not supported on iOS"
-                                 details:nil]);
+    NSString *signInOption = call.arguments[@"signInOption"];
+    if ([signInOption isEqualToString] == @"SignInOption.games") {
```
Comment (mravn-google) [inline packages/google_sign_in/ios/Classes/GoogleSignInPlugin.m]: Oops. You meant `[sigInOption isEqualtoString:@"SignInOption.games"]` :-)

**Comment by mravn-google** [comment]
Published after #519

**Pull request merged**